tests/models/test_NaiveBayes.py

```diff
@@ -0,0 +1,123 @@
+
+        # Split into train/test sets
+        split = train_test_split(X, Y, test_size=0.4)
+        self.X_train = split[0]
+        self.X_test = split[1]
+        self.Y_train = split[2]
+        self.Y_test = split[3]
+
+        self.num_training_samples = self.Y_train.shape[0]
+        self.test_GNB = GaussianNaiveBayes()
+        self.test_dir = tempfile.mkdtemp()
+
+        # Fit the model once and for all
+        self.test_GNB.fit(self.X_train, self.Y_train, minibatch_size=2000)
+
+    def tearDown(self):
+        """
+        Tear down the testing environment.
+        """
+        shutil.rmtree(self.test_dir)
+
+
+class GaussianNaiveBayesFitTestCase(GaussianNaiveBayesTestCase):
+    def test_fit_returns_correct_model(self):
+        """
+        Test the model initialization and fit.
+
+        Currently, only the sign of inferred parameters is checked
+        against the sign of the parameters used to generate the data.
+
+        TOOO: Find better strategies to test probabilistic code.
+        """
+
+        print('')
+
+        # Check that the model correctly infers dimensions
+        self.assertEqual(self.num_cats, self.test_GNB.num_cats)
+        self.assertEqual(self.num_training_samples, self.test_GNB.num_training_samples)
+        self.assertEqual(self.num_pred, self.test_GNB.num_pred)
+
+        # TODO: How do you write tests for a stochastic mode?
+        # TODO: Diagnose the sampling with a reasonable sampling size?
+        np.testing.assert_equal(
+            np.sign(self.pi),
+            np.sign(self.test_GNB.trace['pi'].mean(axis=0))) 
+        np.testing.assert_equal(
+            np.sign(self.sigma),
+            np.sign(self.test_GNB.trace['sigma'].mean(axis=0))
+            )
+
+
+class GaussianNaiveBayesPredictProbaTest(GaussianNaiveBayesTestCase):
+    def test_predict_proba_returns_probabilities(self):
+        print('')
+        probs = self.test_GNB.predict_proba(self.X_test)
+        self.assertEqual(probs.shape[0], self.Y_test.shape[0])
+
+    def test_predict_proba_raises_error_if_not_fit(self):
+        with self.assertRaises(PyMC3ModelsError) as no_fit_error:
+            test_GNB = GaussianNaiveBayes()
+            test_GNB.predict_proba(self.X_train)
+        expected = 'Run fit() on the model before predict()'  # precarious, define a new Exception type
+        self.assertEqual(str(no_fit_error.exception), expected)
+
+
+class GaussianNaiveBayesPredictionTestCase(GaussianNaiveBayesTestCase):
+    def test_predict_returns_predictions(self):
+        """
+        Test that the predict() function's  output has the correct shape.
+        """
+        print('')
+        preds = self.test_GNB.predict(self.X_test)
+        self.assertEqual(preds.shape, self.Y_test.shape)
+
+
+class GaussianNaiveBayesScoreTestCase(GaussianNaiveBayesTestCase):
+    def test_score_scores(self):
+        print('')
+        score = self.test_GNB.score(self.X_test, self.Y_test)
+        # What to test for?
```

Comment:
This is not the most informative. If you're not sure, could you just put in something more like: ```TODO: Figure out how to test the score function```.

Additionally, If there's no test, I would just put use the skip decorator (https://docs.python.org/3/library/unittest.html#unittest.skip): ```@unittest.skip('Not implemented yet')```

There's no reason to run the scoring if there's no test since it just takes up time.

Reply:
You're right!